tools/eslint-rules/snake-case-filenames.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview enforces snake_case for filenames.
+ * @author Shobhit Chittora <chittorashobhit@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const path = require('path');
+
+module.exports = function(context) {
+  const snakeCaseRegex = new RegExp('^[a-z_]+$');
+  const targetDirs = ['lib'];
+
+  function checkIfInTargetDir(dirName) {
```

Comment:
Do not check if you're in the target dir. All this sort of work should be done by the config files. If you really have to do it this way, leave a comment in the code explaining why.